src/zenml/artifact_stores/base_artifact_store.py

```diff
@@ -44,67 +46,126 @@
+class _sanitize_paths:
+    """Sanitizes path inputs before calling the original function.
 
-    If the input is a **remote** path, this function replaces backslash path
-    separators by forward slashes.
+    Extra decoration layer is needed to pass in fixed artifact store root
+    path for static methods that are called on filesystems directly.
 
     Args:
-        potential_path: Value that potentially refers to a (remote) path.
+        func: The function to decorate.
+        fixed_root_path: The fixed artifact store root path.
+        is_static: Whether the function is static or not.
 
     Returns:
-        The original input or a sanitized version of it in case of a remote
-        path.
+        Function that calls the input function with sanitized path inputs.
     """
-    if isinstance(potential_path, bytes):
-        path = fileio.convert_to_str(potential_path)
-    elif isinstance(potential_path, str):
-        path = potential_path
-    else:
-        # Neither string nor bytes, this is not a path
-        return potential_path
-
-    if io_utils.is_remote(path):
-        # If we have a remote path, replace windows path separators with
-        # slashes
-        import ntpath
-        import posixpath
 
-        path = path.replace(ntpath.sep, posixpath.sep)
-
-    return path
+    def __init__(self, func: Any, fixed_root_path: str, is_static: bool):
+        """Initializes the decorator.
 
+        Args:
+            func: The function to decorate.
+            fixed_root_path: The fixed artifact store root path.
+            is_static: Whether the function is static or not.
+        """
+        self.func = func
+        self.fixed_root_path = fixed_root_path
+        self.is_static = is_static
+
+        params = inspect.signature(self.func).parameters
+        self.valid_pos = set()
+        self.valid_key = set()
+        has_self = False
+        for i, k in enumerate(params):
+            if k == "self":
+                has_self = True
+            if params[k].annotation == PathType:
+                self.valid_pos.add(i - has_self)
+                self.valid_key.add(k)
+
+    def _validate_path(
+        self, path: str, pos: Optional[int] = -1, key: Optional[str] = None
+    ) -> None:
+        """Validates the path.
 
-def _sanitize_paths(_func: Callable[..., Any]) -> Callable[..., Any]:
-    """Sanitizes path inputs before calling the original function.
+        Args:
+            path: The path to validate.
+            pos: Position of the path in the function signature.
+            key: Name of the path in the function signature.
 
-    Args:
-        _func: The function for which to sanitize the inputs.
+        Raises:
+            FileNotFoundError: If the path is outside of the artifact store
+                bounds.
+        """
+        if pos in self.valid_pos or key in self.valid_key:
+            if not path.startswith(self.fixed_root_path):
+                raise FileNotFoundError(
```

Comment:
Maybe a `PermissionError` which gives a more accurate sense of the real problem?

Reply:
Here I agree with @stefannica - if the file is not accessible by the artifact store or file system because of anything, it sounds like a `FileNotFound` to me. Rest is unneeded details, imo

Reply:
https://github.com/zenml-io/zenml/pull/2490#discussion_r1509200124